### PR TITLE
feat: use `alloy` for EVM wallet

### PR DESCRIPTION
**Pull request by antbern**
# Pull Request Template

## Motivation and Context

We were using the `iota-wallet` to manually sign and submit transactions to the EVM. This was coumbersome and inefficient. Since we will have to move away from the old iota wallet anyways, this provides a step in the right direction.

## Summary

Provide a short summary of changes in this pull request.

- Replace `iota-wallet` with `alloy` for the Wallet management in EVM impl.
- Update `alloy` crates to latest version (`v0.12.5`)
- Cleanup the `WalletUser` trait to remove duplicated functions and move non-generic functionality out of the trait.
- Update bindings (removed `tag` and `message` for sending transactions, keep only `data: Vec<u8>` for attaching data to each transaction). Updated docs.

## Checklist

Please ensure that your PR meets the following requirements:

### General

- [x] Code follows project style and guidelines.
- [x] No redundant or duplicate code.
- [x] No added new dependencies without clear justification and approval.

### Documentation

- [x] Added/updated relevant documentation (e.g., README, inline comments).
- [ ] Added a CHANGELOG entry for major changes.

### Testing

- [x] Added relevant test cases, leveraging tools like `rstest` or similar for reusable tests.
- [x] Tests pass locally.

### Build & CI/CD

- [x] Confirmed no CI/CD pipeline issues.
- [x] Updated build scripts where necessary.

## Comments to Reviewer

Please provide any comments or points of consideration for the reviewers, such as specific areas of focus, potential edge cases, or context that would aid the review process.

### Reviewer Responsibilities

- [ ] Code adheres to style and guidelines.
- [ ] All tests are appropriately covered and they pass.
- [ ] Consider performance and/or security impacts.
- [ ] Constructive feedback is provided.

### CODE REVIEW

File: sdk/bindings/wasm/examples/utils.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export async function initSdk(username: string) {
     }
     `);
 
-    await sdk.setCurrency(wasm.Currency.Iota);
+    // await sdk.setCurrency(wasm.Currency.Iota);
```
Comment (hauju) [inline sdk/bindings/wasm/examples/utils.ts:37]: Can this be deleted?

**Review by hauju — APPROVED** [review]
Great, just a small comment.

File: sdk/bindings/wasm/examples/utils.ts (hunk 1 above)
Comment (antbern) [reply sdk/bindings/wasm/examples/utils.ts]: for sure, I had forgotten to push my last commit 😆

Comment (antbern) [reply sdk/bindings/wasm/examples/utils.ts]: Reverted to what it was before, will be fixed by #56

File: sdk/bindings/swift/examples/main.swift [hunk 2]
```diff
@@ -0,0 +1,5 @@
+let message = "swift bindings test"
+let data = message.data(using: .utf8)!
+let bytes = [Uint8](data)
+
+let buf: [UInt8] = Array(message.utf8)
```
Comment (kristifidani) [inline sdk/bindings/swift/examples/main.swift:5]: Why do we need this `main.swift` inside the /examples?

**Review by kristifidani — APPROVED** [review]
Simplified a lot, lets go 💯 🚀

**Review by kristifidani — APPROVED** [review]

File: sdk/src/wallet/wallet_user_eth.rs [hunk 3]
```diff
@@ -581,44 +180,51 @@ impl WalletUser for WalletImplEth {
...
+            transaction_id: transaction_id.to_string(),
+            incoming: is_transaction_incoming,
+            amount: value_eth_f64,
+            network: "ETH".to_string(),
+            status: format!("{:?}", status),
+            explorer_url: None,
+        })
     }
 
     async fn estimate_gas_cost_eip1559(&self, transaction: TxEip1559) -> Result<GasCostEstimation> {
```
Comment (lszetg) [inline sdk/src/wallet/wallet_user_eth.rs:230]: since we're using _auto estimate gas_ fillter in `provider`, I think we can remove this method 👀

**Review by lszetg — APPROVED** [review]
I see we're slowly moving away from `tag, data, message` :D 
Nice simplification 🎉

File: sdk/src/wallet/wallet_user_eth.rs (hunk 3 above)
Comment (lszetg) [reply sdk/src/wallet/wallet_user_eth.rs:230]: ah, too late xD

Comment (antbern) [reply sdk/src/wallet/wallet_user_eth.rs:230]: Yeah Sorry 😄 I do agree to some extent, but I think we need to have a generic way for the user to see how much a transaction would cost, hence I left it in. But something like `estimate_gas(to, amount, data) -> CryptoAmount` could be a good start. And for `iota` we can just return `0` for now. But let's do that in a followup, and we also need a bindings etc so I think it can be part of the upcoming wallet / sdk refactorings 💯
